addons/stock_account/models/product.py

```diff
@@ -185,12 +185,22 @@ def _prepare_out_svl_vals(self, quantity, company):
         if self.cost_method in ('average', 'fifo'):
             fifo_vals = self._run_fifo(abs(quantity), company)
             vals['remaining_qty'] = fifo_vals.get('remaining_qty')
-            # in case of AVCO, fix rounding issue of standard price when needed.
+            # In case of AVCO, fix rounding issue of standard price when needed.
             if self.cost_method == 'average':
-                rounding_error = self.standard_price * self.quantity_svl - self.value_svl
-                vals['value'] += self.env.company.currency_id.round(rounding_error)
-                if self.quantity_svl:
-                    vals['unit_cost'] = self.value_svl / self.quantity_svl
+                currency = self.env.company.currency_id
+                rounding_error = currency.round(self.standard_price * self.quantity_svl - self.value_svl)
+                if rounding_error:
+                    # If it is bigger than the (smallest number of the currency * quantity) / 2,
+                    # then it isn't a rounding error but a stock valuation error, we shouldn't fix it under the hood ...
+                    if abs(rounding_error) <= (abs(quantity) * currency.rounding) / 2:
+                        vals['value'] += rounding_error
+                        vals['rounding_adjustment'] = '\nRounding Adjustment: %s%s %s' % (
+                            '+' if rounding_error > 0 else '',
+                            float_repr(rounding_error, precision_digits=currency.decimal_places),
+                            currency.symbol
+                        )
```

Comment:
Isn't there a missing translation here ? :thinking:

FYI, there is a dedicated tool to format an amount in a given currency : `odoo.tools.format_amount`.
I discovered it recently and it's very useful :) (e.g. it will correctly manage currencies whose symbol should be placed before the amount).

(Sorry to be the annoying grumpy guy, I was curious because I'm busy with rounding problems in `payment` too :sweat_smile:)

Reply:
We never translate description on SVL (maybe we should ?). for now, It is for a debug purpose.

I didn't know about the `format_amount`. I will make a small fix about that, I am sure that it isn't the only place that it should be used (in logistics).

Thanks!

Reply:
About the translation, obviously I am wrong, sometime we translate sometime we don't. I will fix that too.